incubator/binding-asyncapi/src/main/java/io/aklivity/zilla/runtime/binding/asyncapi/internal/stream/AsyncapiServerFactory.java

```diff
@@ -0,0 +1,1067 @@
+
+    private final MutableDirectBuffer writeBuffer;
+    private final MutableDirectBuffer extBuffer;
+    private final BufferPool bufferPool;
+    private final BindingHandler streamFactory;
+    private final LongUnaryOperator supplyInitialId;
+    private final LongUnaryOperator supplyReplyId;
+    private final LongSupplier supplyTraceId;
+    private final Long2ObjectHashMap<AsyncapiBindingConfig> bindings;
+    private final int asyncapiTypeId;
+    private final int mqttTypeId;
+    private final AsyncapiConfiguration config;
+
+    public AsyncapiServerFactory(
+        AsyncapiConfiguration config,
+        EngineContext context)
+    {
+        this.config = config;
+        this.writeBuffer = context.writeBuffer();
+        this.extBuffer = new UnsafeBuffer(new byte[writeBuffer.capacity()]);
+        this.bufferPool = context.bufferPool();
+        this.streamFactory = context.streamFactory();
+        this.supplyInitialId = context::supplyInitialId;
+        this.supplyReplyId = context::supplyReplyId;
+        this.supplyTraceId = context::supplyTraceId;
+        this.bindings = new Long2ObjectHashMap<>();
+        this.asyncapiTypeId = context.supplyTypeId(AsyncapiBinding.NAME);
+        this.mqttTypeId = context.supplyTypeId(MQTT_TYPE_NAME);
+    }
+
+    @Override
+    public int originTypeId()
+    {
+        return mqttTypeId;
+    }
+
+    @Override
+    public int routedTypeId()
+    {
+        return asyncapiTypeId;
+    }
+
+    @Override
+    public void attach(
+        BindingConfig binding)
+    {
+        AsyncapiBindingConfig asyncapiBinding = new AsyncapiBindingConfig(binding, config.targetRouteId());
+        bindings.put(binding.id, asyncapiBinding);
+    }
+
+    @Override
+    public void detach(
+        long bindingId)
+    {
+        bindings.remove(bindingId);
+    }
+
+    @Override
+    public MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer receiver)
+    {
+        final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+        final long originId = begin.originId();
+        final long routedId = begin.routedId();
+        final long initialId = begin.streamId();
+        final long affinity = begin.affinity();
+        final long authorization = begin.authorization();
+        final OctetsFW extension = begin.extension();
+        final MqttBeginExFW mqttBeginEx = extension.get(mqttBeginExRO::tryWrap);
+
+        final AsyncapiBindingConfig binding = bindings.get(routedId);
+
+        MessageConsumer newStream = null;
+
+        if (binding != null && binding.isCompositeOriginId(originId))
+        {
+            final AsyncapiRouteConfig route = binding.resolve(authorization);
+
+            if (route != null)
+            {
+                final String operationId = null;
+                newStream = new CompositeStream(
```

Comment:
```suggestion
                final String operationId = null;

                newStream = new CompositeStream(
```
for improved readability